chromium_src/chrome/browser/ui/tabs/tab_renderer_data.cc

```diff
@@ -9,11 +9,31 @@
 #include "src/chrome/browser/ui/tabs/tab_renderer_data.cc"
 #undef FromTabInModel
 
+#include "brave/browser/ui/tabs/features.h"
+#include "brave/browser/ui/tabs/shared_pinned_tab_service.h"
+#include "brave/browser/ui/tabs/shared_pinned_tab_service_factory.h"
 #include "brave/components/constants/webui_url_constants.h"
 #include "url/gurl.h"
 
 TabRendererData TabRendererData::FromTabInModel(TabStripModel* model,
                                                 int index) {
+  if (base::FeatureList::IsEnabled(tabs::features::kBraveSharedPinnedTabs)) {
+    if (index < model->IndexOfFirstNonPinnedTab()) {
+      auto* shared_pinned_tab_service =
+          SharedPinnedTabServiceFactory::GetForProfile(model->GetProfile());
+      DCHECK(shared_pinned_tab_service);
```

Comment:
Hey, just out of interest, why do you `DCHECK` every time you get a service from a factory?

Reply:
That must be my cargo cult about pointer 😆 I seemed to prefer checking pointer validity when there's possibility to return nullity as side effect of future changes. In this case, GetForProfile should return valid service even when the profile is incognito. So that could be what's behind my cult. Sometimes, calling method from nullptr doesn't print proper logs, so I feel like DCHECK()-ing explicitly could be better. But I know I'm not that consistent about this checking 😂

Reply:
Makes sense to me, and it doesn't hurt :laughing: